samples: fs: fat_fs: Add RAM disk sample for nrf52840dk

de-nordic opened the pull request:
Add configuration for nrf52840dk that allows to create RAM disk; this configuration does not reserve special region in RAM using DTS but uses automatic buffer allocation, by RAM Disk Drivers, according to RAM disk specification in DTS.

de-nordic committed:
samples: fs: fat_fs: Add RAM disk sample for nrf52840dk

Add configuration for nrf52840dk that allows to create RAM
disk; this configuration does not reserve special region
in RAM using DTS but uses automatic buffer allocation, by RAM
Disk Drivers, according to RAM disk specification in DTS.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nordicjm commented on samples/subsys/fs/fs_sample/sample.yaml:
```diff
@@ -31,6 +31,12 @@ tests:
     platform_allow: nrf52840dk_nrf52840
+  sample.filesystem.fat_fs.nrf52840dk_nrf52840_ram_disk:
+    build_only: true
+    platform_allow: nrf52840dk_nrf52840
```
Needs rebase and updating for hwmv2

de-nordic committed:
samples: fs: fat_fs: RAM disk in SRAM region for nrf52840dk

Add overlay, for nrf52840dk, that reserves RAM in internal
SRAM using the DTS definition.
The sample with such region can be built with the config
file nrf52840dk_nrf52840_ram_disk_region.conf, but, instead
of automatically allocating memory, the Disk driver will
use the pre-defined region provided by the DTS overlay.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nordicjm reviewed (approved):
Failure in CI is unrelated to change and is an issue with hwmv2, have added it to the issue https://github.com/zephyrproject-rtos/zephyr/issues/70445

danieldegrasse commented on samples/subsys/fs/fs_sample/sample.yaml:
```diff
@@ -31,6 +31,18 @@ tests:
     build_only: true
     platform_allow: nrf52840dk/nrf52840
+  sample.filesystem.fat_fs.nrf52840dk_nrf52840_ram_disk:
+    build_only: true
```
Why build only? (other than that the other testcases already have this set)

de-nordic replied:
I am not sure why we want to run this every time, but we want to know whether it still builds.

danieldegrasse replied:
Fair enough- I guess the runtime overhead is a concern for testing teams?

de-nordic replied:
Yeah, in reality the run should be done by test and verified - that way we will know whether it works internally.
The sample should be just like "Hello world" where we assume that everything works and just show people, in very simplified form, how to do things.
I was asked how do the disk in RAM, several times, and now I can just point to sample.
Building is still taking some CI time, so not perfect anyway.
